src/app/qgsdiagramproperties.cpp

```diff
@@ -119,31 +122,52 @@ QgsDiagramProperties::QgsDiagramProperties( QgsVectorLayer *layer, QWidget *pare
     mDiagramFrame->setEnabled( false );
   }
 
-  //insert placement options
-  mPlacementComboBox->blockSignals( true );
-  switch ( layerType )
+  // set placement methods page based on geometry type
+
+  switch ( mLayer->geometryType() )
   {
     case QgsWkbTypes::PointGeometry:
-      mPlacementComboBox->addItem( tr( "Around Point" ), QgsDiagramLayerSettings::AroundPoint );
-      mPlacementComboBox->addItem( tr( "Over Point" ), QgsDiagramLayerSettings::OverPoint );
+      stackedPlacement->setCurrentWidget( pagePoint );
       mLinePlacementFrame->setVisible( false );
       break;
     case QgsWkbTypes::LineGeometry:
-      mPlacementComboBox->addItem( tr( "Around Line" ), QgsDiagramLayerSettings::Line );
-      mPlacementComboBox->addItem( tr( "Over Line" ), QgsDiagramLayerSettings::Horizontal );
+      stackedPlacement->setCurrentWidget( pageLine );
       mLinePlacementFrame->setVisible( true );
       break;
     case QgsWkbTypes::PolygonGeometry:
-      mPlacementComboBox->addItem( tr( "Around Centroid" ), QgsDiagramLayerSettings::AroundPoint );
-      mPlacementComboBox->addItem( tr( "Over Centroid" ), QgsDiagramLayerSettings::OverPoint );
-      mPlacementComboBox->addItem( tr( "Perimeter" ), QgsDiagramLayerSettings::Line );
-      mPlacementComboBox->addItem( tr( "Inside Polygon" ), QgsDiagramLayerSettings::Horizontal );
+      stackedPlacement->setCurrentWidget( pagePolygon );
       mLinePlacementFrame->setVisible( false );
       break;
-    default:
+    case QgsWkbTypes::NullGeometry:
       break;
+    case QgsWkbTypes::UnknownGeometry:
+      qFatal( "unknown geometry type unexpected" );
   }
-  mPlacementComboBox->blockSignals( false );
+
+  //insert placement options
+  // setup point placement button group
+  mPlacePointBtnGrp = new QButtonGroup( this );
+  mPlacePointBtnGrp->addButton( radAroundPoint, ( int )QgsDiagramLayerSettings::AroundPoint );
```

Comment:
Is the `(int)` cast required here? Try to remove it.

Reply:
I try to revert the logic and remove all these mPlaceBtnGrp and the connect directly to the button. It led me to something like this
```
  QWidget *curWdgt = stackedPlacement->currentWidget();
  QButtonGroup *rgroup = curWdgt->findChild<QButtonGroup *>();
  connect( rgroup, static_cast<void ( QButtonGroup::* )( int )>( &QButtonGroup::buttonClicked ), this, &QgsDiagramProperties::updatePlacementWidgets );
```
But it doesn't seem to return the clicked button. What's wrong here? Any help to move forward would be very welcome. thanks

Reply:
I had tested with returnin a list of radiobuttons children (from memory)
`  QRadioButton *rgroup = curWdgt.findChildren<QButtonGroup *>();`
 but i don't know how to write the connect command based on the returned list

Reply:
Sorry, I think I'm missing the point here. My proposal was just to change lines like this

```cpp
mPlacePointBtnGrp->addButton( radAroundPoint, ( int )QgsDiagramLayerSettings::AroundPoint );
```
to

```cpp
mPlacePointBtnGrp->addButton( radAroundPoint );
```

Then I would just connect to the button groups `buttonToggled` signal, loop over the buttons to see which one `isChecked` - just like it's done in `updatePlacementWidgets()`.

Reply:
Ok. I think I see what you propose. Will give it a try (should be easier than what I was trying). 
What I wanted to do is to keep the `updatePlacementWidgets` as it was before, using the `QgsDiagramLayerSettings::Placement currentPlacement` variable instead of those long conditions based on the radio buttons naming.
I will push the whole code later, when I have access to the computer.